### ISSUE TITLE
Should dif.do be based on measurement or model DO?

### ISSUE THREAD

**Issue by jread-usgs** [GLEON/LakeMetabolizer#93]
See relevant discussion over here at #streamMetabolizer https://github.com/USGS-R/streamMetabolizer/pull/25#discussion_r29093608

**Comment by jread-usgs** [comment]
relevant code: https://github.com/GLEON/LakeMetabolizer/blob/master/R/metab.bookkeep.R#L102
and https://github.com/USGS-R/streamMetabolizer/blob/master/R/metab_simple.R#L69

**Comment by rBatt** [comment]
I gave an answer on that discussion you linked originally.

We have no choice in metab.bookkeep – there are no predicted values. There
aren't any error terms.

For me, it gets kinda fuzzy and weird for metab.ols. To do the regression,
we have to calculate gas flux using just the observations, and we actually
model our response variable as the first difference of DO observations
minus an offset for gas exchange. That's why I've considered the response
variable in metab.ols as "NEP", rather than saying NEP is the sum of GPP
and R (because of the error term, these aren't always the same).

For metab.mle everything should be done with the modeled values. It's hard
to tell just by looking at our code b/c we solved the differential equation
to do a better job of modeling how DO would change due to gas flux between
long time steps (thanks Gordon), but we do use the modeled value of DO in
the gas flux (which is why we solve the diff anyway).

metab.kalman and metab.bayesian do everything with the modeled value,
except those models are useful to look at because they are very explicit
about distinguishing "predictions" from "observations" from "true values"

Reading that I've found helpful for some of this:
http://ms.mcmaster.ca/~bolker/emdbook/chap11A.pdf
